plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -173,19 +171,20 @@ private void validateCatalogs(List<CatalogMetadata> catalogs) {
             + " Allowed characters: a-zA-Z0-9_-*@ ", catalog.getName()));
       }
 
-      if (StringUtils.isEmpty(catalog.getUri())) {
-        LOG.error("Found a catalog with no uri. {}", catalog.toString());
-        throw new IllegalArgumentException(
-            "Missing URI Field from a catalog. URI is a required parameter.");
-      }
-
       String catalogName = catalog.getName();
       if (reviewedCatalogs.contains(catalogName)) {
         LOG.error("Found duplicate catalog names");
         throw new IllegalArgumentException("Catalogs with same name are not allowed.");
       } else {
         reviewedCatalogs.add(catalogName);
       }
+
+      if (Objects.isNull(catalog.getProperties())) {
+        LOG.error("Found a catalog with no properties. {}", catalog.toString());
+        throw new IllegalArgumentException("Missing properties field in catalog configuration. "
+            + "Properties are required parameters");
```

Comment:
Just want to confirm where is this exception captured and handled finally? Would it impact the entire cluster if user configures an invalid catalog?

Reply:
Thanks, this is a good catch. 
Yes, the cluster will be impacted it will be down with errors.
1) If the user doesn't configure anything, it won't fail.
2) If he configures something and has given wrong configuration, the cluster will fail to start. 

Should I just log the error and don't throw any exception?
Only Logging error without exception would make sense, because now I am providing a way to reload the catalog configuration even while the cluster is running

Reply:
Any catalog configuration error will only lead to non-registration of catalog and error logs. The cluster should work fine.